gf/include/alps/gf/mesh.hpp

```diff
@@ -659,7 +659,124 @@ namespace alps {
-    
+
+        class legendre_mesh : public base_mesh {
+            double beta_;
+            int n_max_;//number of legendre polynomials
+
+            statistics::statistics_type statistics_;
+
+        public:
+            typedef generic_index<legendre_mesh> index_type;
+
+            legendre_mesh(gf::statistics::statistics_type statistics=statistics::FERMIONIC):
+                    beta_(0.0), n_max_(0), statistics_(statistics) {}
+
+            legendre_mesh(double b, int n_max, gf::statistics::statistics_type statistics=statistics::FERMIONIC):
+                    beta_(b), n_max_(n_max), statistics_(statistics) {
+                check_range();
+                compute_points();
+            }
+            int extent() const{return n_max_;}
+
+            int operator()(index_type idx) const {
+                return idx();
+            }
+
+            /// Comparison operators
+            bool operator==(const legendre_mesh &mesh) const {
+                return beta_==mesh.beta_ && n_max_==mesh.n_max_ && statistics_==mesh.statistics_;
+            }
+
+            /// Comparison operators
+            bool operator!=(const legendre_mesh &mesh) const {
+                return !(*this==mesh);
+            }
+
+            ///getter functions for member variables
+            double beta() const{ return beta_;}
+            statistics::statistics_type statistics() const{ return statistics_;}
+
+            /// Swaps this and another mesh
+            // It's a member function to avoid dealing with templated friend decalration.
+            void swap(legendre_mesh& other) {
+                using std::swap;
+                if (this->statistics_ != other.statistics_) {
+                    throw std::runtime_error("Attempt to swap LEGENDRE meshes with different statistics");
+                }
+                swap(this->beta_, other.beta_);
+                swap(this->n_max_, other.n_max_);
+                base_mesh::swap(other);
+            }
+
+            void save(alps::hdf5::archive& ar, const std::string& path) const
+            {
+                ar[path+"/kind"] << "LEGENDRE";
+                ar[path+"/N"] << n_max_;
+                ar[path+"/statistics"] << int(statistics_);
+                ar[path+"/beta"] << beta_;
+            }
+
+            void load(alps::hdf5::archive& ar, const std::string& path)
+            {
+                std::string kind;
+                ar[path+"/kind"] >> kind;
+                if (kind!="LEGENDRE") throw std::runtime_error("Attempt to read LEGENDRE mesh from non-LEGENDRE data, kind="+kind); // FIXME: specific exception
+                double n_max, beta;
+                int stat;
+
+                ar[path+"/N"] >> n_max;
+                ar[path+"/statistics"] >> stat;
+                ar[path+"/beta"] >> beta;
+
+                statistics_=statistics::statistics_type(stat);
+                beta_=beta;
+                n_max_=n_max;
+                check_range();
```

Comment:
Sure. :)  This is not the only place where we do not offer a strong exception guarantee. :(

But this can be addressed later and everywhere at once.

As i said, it is probably not a big deal in practice, so i am going to accept the merge.

Hiroshi (@shinaoka) and Sergey (@iskakoff), thanks for the contribution and the review!